DataDog/datadog-operator
# create Pod Disruption Budget for DCA and CCR deployments

### What does this PR do?

Create Pod Disruption budgets for cluster agent and cluster checks runners.

### Motivation

#864 We had PDB in version 0.8.x but it was not included in later versions of the operator.

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

- create a kind cluster
- in `datadog-agent.yaml`:
    - enable PDB for DCA and CLC -- override.ClusterAgent.createPodDisruptionBudget = true
    - add replicas:  override.ClusterAgent.replicas = 3
- deploy operator and agent to your kind cluster
- run `kubectl get poddisruptionbudgets` and you should see the DCA and CLC PDBs created. CLC has a maxUnavailable of 1 pod, and DCA has a minAvailable of 1 pod.
- run `kubectl get pods -o wide` to see which node the DCA and CLC pods are on. 
- run `k drain <node> --ignore-daemonsets --delete-emptydir-data`. You should get a message like this: `Cannot evict pod as it would violate the pod's disruption budget.`
- run `kubectl get pods -o wide` to see the status of the nodes now. There should only be one running pod for DCA, and only one pending pod for CLC. 
- run `kubectl uncordone <node>`
- run `kubectl get pods -o wide`-- everything should be back to running.
- remove the flags for PDB and reinstall the agent -- there are no more PDBs created as it is not enabled by default. If you drain a node without PDB, you shouldn't see any error statements and all the pods on that node will have a status of Pending until the node is uncordoned.

### Checklist
